src/Controls/tests/BindingSourceGen.UnitTests/IntegrationTests.cs

```diff
@@ -111,6 +111,338 @@ private static bool ShouldUseSetter(BindingMode mode, BindableProperty bindableP
+                    [InterceptsLocationAttribute(@"Path\To\Program.cs", 5, 7)]
+                    public static void SetBinding1(
+                        this BindableObject bindableObject,
+                        BindableProperty bindableProperty,
+                        Func<global::MyNamespace.A, int> getter,
+                        BindingMode mode = BindingMode.Default,
+                        IValueConverter? converter = null,
+                        object? converterParameter = null,
+                        string? stringFormat = null,
+                        object? source = null,
+                        object? fallbackValue = null,
+                        object? targetNullValue = null)
+                    {
+                        Action<global::MyNamespace.A, int>? setter = null;
+                        if (ShouldUseSetter(mode, bindableProperty))
+                        {
+                            setter = static (source, value) =>
+                            {
+                                if (source is {} p0
+                                    && p0.B is {} p1)
+                                {
+                                    p1.C = value.Value;
+                                }
+                            };
+                        }
+
+                        var binding = new TypedBinding<global::MyNamespace.A, int>(
+                            getter: source => (getter(source), true),
+                            setter,
+                            handlers: new Tuple<Func<global::MyNamespace.A, object?>, string>[]
+                            {
+                                new(static source => source, "B"),
+                                new(static source => source?.B, "C"),
+                            })
+                        {
+                            Mode = mode,
+                            Converter = converter,
+                            ConverterParameter = converterParameter,
+                            StringFormat = stringFormat,
+                            Source = source,
+                            FallbackValue = fallbackValue,
+                            TargetNullValue = targetNullValue
+                        };
+                        bindableObject.SetBinding(bindableProperty, binding);
+                    }
+
+                    private static bool ShouldUseSetter(BindingMode mode, BindableProperty bindableProperty)
+                        => mode == BindingMode.OneWayToSource
+                            || mode == BindingMode.TwoWay
+                            || (mode == BindingMode.Default
+                                && (bindableProperty.DefaultBindingMode == BindingMode.OneWayToSource
+                                    || bindableProperty.DefaultBindingMode == BindingMode.TwoWay));
+                }
+            }
+            """,
+            result.GeneratedCode);
+    }
+
+    [Theory]
+    [InlineData(
```

Comment:
I don't think this is a good use-case for `InlineData`, I suggest using `MemberData` instead.